typescript/deploy/src/router/check.ts

```diff
@@ -14,27 +16,33 @@ import { RouterConfig } from './types';
 
 export class AbacusRouterChecker<
   Chain extends ChainName,
-  App extends AbacusApp<RouterContracts, Chain>,
-  Config extends RouterConfig,
-> extends AbacusAppChecker<Chain, App, Config> {
+  Contracts extends RouterContracts,
+  App extends AbacusApp<Contracts, Chain>,
+  Config,
+> extends AbacusAppChecker<Chain, App, Config & RouterConfig> {
   checkOwnership(chain: Chain) {
     const owner = this.configMap[chain].owner;
     const ownables = this.ownables(chain);
     return AbacusAppChecker.checkOwnership(owner, ownables);
   }
 
+  getRouterInstance(router: Router | ProxiedContract<Router, any>) {
+    return router instanceof ProxiedContract ? router.contract : router;
```

Comment:
Happy to add it. Does that still apply if one of the objects is not "ours", but an ethers object?

Reply:
idk if I agree
in this case instanceof feels more robust

Reply:
The problem general concerns with `instanceOf` are that it doesn't work across frames (probably not relevant here) and that it can break in some more advanced OOP cases (b.c. it relies on constructor identity comparison).

I'm probably prematurely optimizing here so if @yorhodes and you prefer `instanceOf` feel free to ignore. Just would recommend against this becoming a common pattern in the SDK